aws-event-bridge/src/test/scala/akka/stream/alpakka/aws/eventbridge/EventBridgePublishMockSpec.scala

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.aws.eventbridge
+
+import java.util.concurrent.CompletableFuture
+
+import akka.stream.alpakka.aws.eventbridge.scaladsl.EventBridgePublisher
+import akka.stream.scaladsl.{Keep, Sink}
+import akka.stream.testkit.scaladsl.TestSource
+import org.mockito.ArgumentMatchers.{any, eq => meq}
+import org.mockito.Mockito._
+import org.scalatest._
+import software.amazon.awssdk.services.eventbridge.model._
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class EventBridgePublishMockSpec extends FlatSpec with DefaultTestContext with MustMatchers {
+
+  private def entryDetail(detail: String): PutEventsRequestEntry =
+    PutEventsRequestEntry.builder().detail(detail).build()
+
+  private def requestDetail(detail: String): PutEventsRequest =
+    PutEventsRequest.builder().entries(entryDetail(detail)).build()
+
+  private def resultResponse(eventId: String): PutEventsResponse = {
+    val putResultEntry = PutEventsResultEntry.builder().eventId(eventId).build()
+    PutEventsResponse.builder().entries(putResultEntry).build()
+  }
+
+  it should "publish a single PublishRequest message to eventBridge" in {
+    val putRequestEntry = PutEventsRequestEntry.builder().detail("event-bridge-message").build()
+    val putRequest = requestDetail("event-bridge-message")
+    val putResult = resultResponse("event-id")
+
+    when(eventBridgeClient.putEvents(putRequest)).thenReturn(CompletableFuture.completedFuture(putResult))
+
+    val (probe, future) =
+      TestSource.probe[PutEventsRequest].via(EventBridgePublisher.publishFlow()).toMat(Sink.seq)(Keep.both).run()
+
+    probe.sendNext(PutEventsRequest.builder().entries(putRequestEntry).build()).sendComplete()
+
+    Await.result(future, 1.second) mustBe putResult :: Nil
+
+    verify(eventBridgeClient, times(1)).putEvents(meq(putRequest))
+  }
+
+  it should "publish multiple PutEventsRequest messages to eventbridge" in {
+    val putResultEntry = PutEventsResultEntry.builder().eventId("event-id").build()
+    val putResult = PutEventsResponse.builder().entries(putResultEntry).build()
```

Comment:
Can this be DRYd up with your `resultResponse` method?  I think there are a few other instances in this test too.